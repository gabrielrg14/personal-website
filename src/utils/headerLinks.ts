const headerLinks = [
    {
        href: "/about",
        label: "About me",
    },
    {
        href: "/projects",
        label: "Projects",
    },
];
  
export default headerLinks;