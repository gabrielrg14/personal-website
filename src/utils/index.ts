export * from "./links"
export * from "./mediaQuery"
export * from "./repositoryName"
export * from "./techs"
