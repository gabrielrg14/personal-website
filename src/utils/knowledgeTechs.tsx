import {
    Vuedotjs,
    Redux,
    Nodedotjs,
    Express,
    Wordpress,
} from "@styled-icons/simple-icons/";
  
const knownTechs = [
    {
        icon: <Vuedotjs />,
        label: "Vue.js",
    },
    {
        icon: <Redux />,
        label: "Redux",
    },
    {
        icon: <Nodedotjs />,
        label: "Node.js",
    },
    {
        icon: <Express />,
        label: "Express.js",
    },
    {
        icon: <Wordpress />,
        label: "Wordpress",
    },
];
  
export default knownTechs;