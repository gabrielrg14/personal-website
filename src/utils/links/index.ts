export * from "./headerLinks"
export * from "./socialMedias"
