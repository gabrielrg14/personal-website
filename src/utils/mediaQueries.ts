import { generateMedia } from "styled-media-query";

const mediaQueries = generateMedia({
    mobile: "576px",
    tablet: "768px",
    desktop: "1200px",
    largeDesktop: "1400px",
});

export default mediaQueries;