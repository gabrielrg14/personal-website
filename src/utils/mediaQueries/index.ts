import { generateMedia } from "styled-media-query"

export const mediaQueries = generateMedia({
    mobile: "576px",
    tablet: "768px",
    desktop: "1200px",
    largeDesktop: "1400px"
})
