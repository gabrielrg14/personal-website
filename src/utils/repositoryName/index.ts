export const formatRepositoryName = (repositoryName: string) => {
    // Removes hyphen from repository name and puts a blank in its place
    return (
        repositoryName
            .replace(/-/g, " ")
            // Capitalize the first letter of each word that makes up the repository's name
            .replace(/(^\w{1})|(\s+\w{1})/g, (letter) => letter.toUpperCase())
    )
}
