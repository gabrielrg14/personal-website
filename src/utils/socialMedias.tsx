import { 
    Linkedin,
    Github,
} from "@styled-icons/simple-icons/";

const socialMedias = [
    {
        link: "https://www.linkedin.com/in/gabrielrapuccigonzalez/",
        icon: <Linkedin />,
    },
    {
        link: "https://github.com/gabrielrg14",
        icon: <Github />,
    },
];

export default socialMedias;