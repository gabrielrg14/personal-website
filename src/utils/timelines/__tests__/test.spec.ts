import { experiences } from "utils"

describe("timelines", () => {
    describe("experiences", () => {
        it("should match snapshot", () => {
            expect(experiences).toMatchSnapshot()
        })

        it("should have a length greater than 0", () => {
            expect(experiences.length).toBeGreaterThan(0)
        })

        it("should contain all properties", () => {
            expect(experiences[0]).toHaveProperty("title")
            expect(experiences[0]).toHaveProperty("company")
            expect(experiences[0]).toHaveProperty("period")
            expect(experiences[0]).toHaveProperty("description")
            expect(experiences[0]).toHaveProperty("skills")
        })

        it("should have the skills property with a length greater than 0", () => {
            expect(experiences[0].skills.length).toBeGreaterThan(0)
        })
    })
})
