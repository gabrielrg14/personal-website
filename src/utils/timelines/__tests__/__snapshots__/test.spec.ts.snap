// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timelines experiences should match snapshot 1`] = `
[
  {
    "company": "Allergisa, a business of ALS",
    "description": "Responsible for maintaining, carrying out updates/corrections and creating new features for the company's internal system, through the SCRUM work methodology.",
    "period": "May 2021 - Present",
    "skills": [
      "react",
      "javascript",
      "bootstrap",
      "angularjs",
      "git",
    ],
    "title": "Front-end Developer",
  },
  {
    "company": "Lenscope",
    "description": "Responsible for developing new landing pages for lens e-commerce, maintaining and updating pages and products created and also monitoring hosting directly on the provider's service, implementing security and performance improvements for the website.",
    "period": "Oct 2019 - Oct 2020",
    "skills": [
      "html",
      "css",
      "javascript",
      "jquery",
      "bootstrap",
      "php",
      "prestashop",
      "wordpress",
      "mysql",
      "git",
    ],
    "title": "Web Development Intern",
  },
]
`;
